packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -28,6 +28,101 @@ describe("publish", () => {
     unsetAllMocks();
   });
 
+  describe("environments", () => {
+    describe("legacy", () => {
+      it("uses the script name", async () => {
+        writeWranglerToml();
+        writeWorkerSource();
+        mockSubDomainRequest();
+        mockUploadWorkerRequest({
+          legacyEnv: true,
+        });
+        await runWrangler("publish index.js --legacy-env true");
+        expect(std.out).toMatchInlineSnapshot(`
+          "Uploaded
+          test-name
+          (TIMINGS)
+          Published
+          test-name
+          (TIMINGS)
+
+          test-name.test-sub-domain.workers.dev"
+        `);
+        expect(std.err).toMatchInlineSnapshot(`""`);
+        expect(std.warn).toMatchInlineSnapshot(`""`);
+      });
+
+      it("appends the environment name when provided", async () => {
+        writeWranglerToml();
+        writeWorkerSource();
+        mockSubDomainRequest();
+        mockUploadWorkerRequest({
+          env: "some-env",
+          legacyEnv: true,
+        });
+        await runWrangler("publish index.js --env some-env --legacy-env true");
+        expect(std.out).toMatchInlineSnapshot(`
+          "Uploaded
+          test-name-some-env
+          (TIMINGS)
+          Published
+          test-name-some-env
+          (TIMINGS)
+
+          test-name-some-env.test-sub-domain.workers.dev"
+        `);
+        expect(std.err).toMatchInlineSnapshot(`""`);
+        expect(std.warn).toMatchInlineSnapshot(`""`);
+      });
+    });
+    describe("services", () => {
+      it("uses the script name", async () => {
+        writeWranglerToml();
+        writeWorkerSource();
+        mockSubDomainRequest();
+        mockUploadWorkerRequest({
+          legacyEnv: false,
+        });
+        await runWrangler("publish index.js");
+        expect(std.out).toMatchInlineSnapshot(`
+          "Uploaded
+          test-name
+          (TIMINGS)
+          Published
+          test-name
+          (TIMINGS)
+
+          test-name.test-sub-domain.workers.dev"
+        `);
+        expect(std.err).toMatchInlineSnapshot(`""`);
+        expect(std.warn).toMatchInlineSnapshot(`""`);
+      });
+
+      it("publishes as an environment when provided", async () => {
+        writeWranglerToml();
+        writeWorkerSource();
+        mockSubDomainRequest();
+        mockUploadWorkerRequest({
+          env: "some-env",
+          legacyEnv: false,
+        });
+        await runWrangler("publish index.js --env some-env");
+        expect(std.out).toMatchInlineSnapshot(`
+          "Uploaded
+          test-name (some-env)
+          (TIMINGS)
+          Published
+          test-name (some-env)
+          (TIMINGS)
+
+          some-env.test-name.test-sub-domain.workers.dev"
```

Comment:
The main difference between legacy and non-legacy appears to be that instead of the worker being deployed as `<script>-<environment>...` it is now `<environment>.<script>...`, right?

Reply:
Correct